test/e2e/specs/editor/various/list-view.spec.js

```diff
@@ -0,0 +1,345 @@
+	// Wait until the current item is selected to ensure UI is ready.
+	await page
+		.getByRole( 'gridcell', {
+			name: /link/i,
+			selected: true,
+		} )
+		.waitFor();
+};
+
+test.describe( 'List View', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'allows a user to drag a block to a new sibling position', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Insert a couple of blocks of different types.
+		await editor.insertBlock( { name: 'core/heading' } );
+		await editor.insertBlock( { name: 'core/image' } );
+		await editor.insertBlock( { name: 'core/paragraph' } );
+
+		// Open the List View panel.
+		await openListView( pageUtils );
+
+		// Ensure the setup is correct before dragging.
+		await expect
+			.poll( editor.getBlocks )
+			.toMatchObject( [
+				{ name: 'core/heading' },
+				{ name: 'core/image' },
+				{ name: 'core/paragraph' },
+			] );
+
+		// Drag the paragraph above the heading.
+		const paragraphBlockItem = page.getByRole( 'gridcell', {
+			name: 'Paragraph link',
+		} );
+		const headingBlockItem = page.getByRole( 'gridcell', {
+			name: 'Heading link',
+		} );
+		await paragraphBlockItem.dragTo( headingBlockItem, { x: 0, y: 0 } );
+
+		// Ensure the block was dropped correctly.
+		await expect
+			.poll( editor.getBlocks )
+			.toMatchObject( [
+				{ name: 'core/paragraph' },
+				{ name: 'core/heading' },
+				{ name: 'core/image' },
+			] );
+	} );
+
+	// Check for regressions of https://github.com/WordPress/gutenberg/issues/38763.
+	test( 'shows the correct amount of blocks after a block is removed in the canvas', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Insert a couple of blocks of different types.
+		await editor.insertBlock( { name: 'core/image' } );
+		await editor.insertBlock( { name: 'core/heading' } );
+		await editor.insertBlock( { name: 'core/paragraph' } );
+
+		// Open the List View panel.
+		await openListView( pageUtils );
+
+		// Go to the image block in List View.
+		await pageUtils.pressKeyTimes( 'ArrowUp', 2 );
+		await expect(
+			page
+				.getByRole( 'gridcell', {
+					name: 'Image link',
+				} )
+				.locator( 'a' )
+		).toBeFocused();
+
+		// Select the image block in the canvas.
+		await page.keyboard.press( 'Enter' );
+		const imageBlock = page.getByRole( 'document', {
+			name: 'Block: Image',
+		} );
+		await expect(
+			imageBlock.getByRole( 'button', { name: 'Upload' } )
+		).toBeFocused();
+		await page.keyboard.press( 'ArrowUp' );
+		await expect( imageBlock ).toBeFocused();
+
+		// Start listeninig to console errors.
+		let hasThrownConsoleError = false;
+		page.on( 'console', ( msg ) => {
+			if ( msg.type() === 'error' ) {
+				hasThrownConsoleError = true;
+			}
+		} );
+
+		// Delete the image block in the canvas.
+		await page.keyboard.press( 'Backspace' );
+
+		// List View should have two rows.
+		await expect(
+			page.getByRole( 'gridcell', { name: /link/i } )
```

Comment:
Nit: I think we can just do string literal here:

```suggestion
			page.getByRole( 'gridcell', { name: 'link' } )
```

Same for other places below.

Reply:
I think we're not yet at Playwright 1.28, so this would be an [exact match](https://playwright.dev/docs/next/api/class-framelocator#frame-locator-get-by-role-option-exact)? This regex is intended because those `gridcell` names contain the block name as well, e.g. `Image link`, `Header link`, etc. and I want to locate them all.

Reply:
I think it would still by default match substring, or it will be a breaking change. I didn't test it though, maybe I'm wrong 😅. Feel free to ignore :)

Reply:
It does the case-insensitive match but doesn't substring. I've just double-checked and got this:

```
locator.waitFor: Timeout 10000ms exceeded.
=========================== logs ===========================
waiting for selector "role=gridcell[selected=true][name="link"i]" to be visible
============================================================
```